extension/extensionlimiter/README.md

```diff
@@ -0,0 +1,352 @@
+
+    // Request items and memory size are applied in the pipeline.
+    return s.nextMetrics.ConsumeMetrics(ctx, data)
+}
+```
+
+#### gRPC stream receiver
+
+A gRPC streaming receiver that holds memory across its allocated in
+`Send()` and does not release it until after a corresponding `Recv()`
+requires use of the lower-level `ResourceLimiter` interface.
+The gRPC  config object's `middlewares` field
+automatically configures network bytes and request count limits:
+
+```yaml
+receivers:
+  streamer:
+    grpc:
+      middlewares:
+      - ratelimiter/streamer
+```
+
+The receiver will check `s.checker.MustDeny()` as above.  In a stream,
+limiters are expected to block the stream until limit requests
+succeed, however after the limit requests succeed, the receiver may
+wish to return from `Send()` to continue accepting new requests while
+the consumer works in a separate goroutine. The limit will be released
+after the consumer returns.
+
+```golang
+func (s *scraper) LogsStream(ctx context.Context, stream *Stream) error {
+    for {
+        // Check saturation for all limiters, all keys.
+        err := s.checker.MustDeny(ctx)
+        if err != nil { ... }
+
+        // The network bytes and request count limits are applied in middleware.
+        req, err := stream.Recv()
+        if err != nil { ... }
+
+        // Allocate memory objects.
+        data, err := s.getLogs(ctx, req)
+        if err != nil { ... }
+
+        release, err := s.memorySizeLimiter.Acquire(ctx, pdataSize(data))
+        if err != nil { ... }
+
+        go func() {
+            // Request items limit is applied in the pipeline consumer
+            err := s.nextMetrics.ConsumeMetrics(ctx, data)
+
+            // Release the memory.
+            release()
+
+            // Reply to the caller.
+            stream.Send(streamResponseFromConsumerError(err))
+        }
+    }
+}
+```
+
+#### Open questions
+
+##### Middleware implementation details
+
+Details are
+important. [#12700](https://github.com/open-telemetry/opentelemetry-collector/pull/12700)
+contained a `limitermiddleware` implementation which was a middleware
+that called a limiter for HTTP and gRPC. Roughly the same code will be
+used, and the details will come out.
+
+##### Provider options
+
+An `Option` type has been added as a placeholder in the provider
+interfaces. **NOTE: No options are implemented.** Potential options:
+
+- The protocol name
+- The signal kind
+- The caller's component ID
+
+Because the set of each of these is small, it is possible to
+pre-compute limiter instances for the cross product of configurations.
+
+##### Context-dependent limits
+
+Client metadata (i.e., headers) may be used in the context to make
+limiter decisions. These details are automatically extracted from the
+Context passed to `MustDeny`, `Limit`, `Acquire`, and `LimitCall`
+functions. No examples are provided. How will limiters configure, for
+example, tenant-specific limits?
+
+##### Data-dependent limits
```

Comment:
If [#39199](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/39199) is accepted, do you still need support here?